### PR TITLE
Add spotlights `002-step-probes` and `004-xdsl`

### PR DESCRIPTION
**Pull request by Haleshot**
Week 2: _August 15, 2024_

Adding community spotlights from Georgios Varnavides and his `stem-probes` file contents.

### CODE REVIEW

File: 002-stem-probes/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+You can explore Georgios' notebook locally with:
+
+```shell
+uvx marimo edit --sandbox stem-probes.py
+```
+
+if you have `uv` installed, or
+
+```shell
+marimo run --sandbox 002-stem-probes\stem-probes.py
```
Comment (akshayka) [inline 002-stem-probes/README.md]: Should we recommend `marimo edit` as a general rule?

Additionally, let's use POSIX-style paths with forward slashes.

```suggestion
marimo edit --sandbox 002-stem-probes/stem-probes.py
```

**Review by akshayka — COMMENTED** [review]
This looks fantastic, thank you!

I left one small comment. Additionally, in the top-level README, please add a bullet for this entry, something like

2. Georgios Varnavides: Miller Fellow at UC Berkeley

Otherwise I think it's great. I love the open with marimo badge, great idea!

File: 002-stem-probes/README.md (hunk 1 above)
Comment (Haleshot) [reply 002-stem-probes/README.md]: I was pondering on whether to use `edit` or `run`; as these notebooks are meant to be run; they are finished programs showcasing various implementations using marimo.

Sure, will change it to `edit` as the user can always toggle apple view to see the outputs.

**Comment by Haleshot** [comment]
Week 4: _August 29, 2024_
Added `004-xdsl` related file contents.

File: 002-stem-probes/README.md (hunk 1 above)
Comment (akshayka) [reply 002-stem-probes/README.md]: @Haleshot , that makes sense. My concern was that in the past users saw `marimo run` and didn't know how to edit the notebook (they thought they had to edit the .py file directly). We can recommend both.

**Review by akshayka — APPROVED** [review]
Wow, fantastic work @Haleshot! I love the assets you've added to the README.

**Pull request merged**